Azdevify contoso-chat

zedy-wj opened the pull request:
Main features:

1. Provisioning via `azd`. (Complete)
2. Add hooks to complete connections. (Complete)
3. Deploying via `azd`. (Not clear, waiting for reply)

@jongio - for notification.

jongio commented on azure.yaml:
```diff
@@ -1,14 +1,11 @@
-workflow:
-  up:
-    - azd: provision
+  template: contoso-chat@0.0.1-beta
```
Please keep the workflow

jongio commented on azure.yaml:
```diff
@@ -1,14 +1,11 @@
-    interactive: true
-    run: infra/hooks/postprovision.sh
\ No newline at end of file
+    interactive: false
+    run: ./infra/hooks/prepenv.sh
```
Let's keep the postprovision file name and you don't need to start with ./

jongio commented on connections/create-connections.py:
```diff
@@ -0,0 +1,61 @@
+import os
```
See my postprovision.sh file to see how I'm doing this with jypter - We don't want to have the code duplicated in the repo.

jongio commented on infra/app/ml.bicep:
```diff
@@ -0,0 +1,119 @@
+param appInsightId string
+param containerRegistryId string
+param contosoChatSfAIName string = 'contoso-chat-sf-ai'
```
```suggestion
param contosoChatSfAiName string = 'contoso-chat-sf-ai'
```

jongio commented on infra/app/ml.bicep:
```diff
@@ -0,0 +1,119 @@
+param storageAccountId string
+
+// In ai.azure.com: Azure AI Resource
+resource mlHub 'Microsoft.MachineLearningServices/workspaces@2023-08-01-preview' = {
```
```suggestion
resource workspace 'Microsoft.MachineLearningServices/workspaces@2023-08-01-preview' = {
```

jongio commented on infra/app/ml.bicep:
```diff
@@ -0,0 +1,119 @@
+    v1LegacyMode: false
+    containerRegistry: containerRegistryId
+    publicNetworkAccess: 'Enabled'
+    discoveryUrl: 'https://swedencentral.api.azureml.ms/discovery'
```
Seems like this should be the location of the resource, not hardcoded to swedencentral

jongio commented on infra/app/ml.bicep:
```diff
@@ -0,0 +1,119 @@
+}
+
+// In ai.azure.com: Azure AI Project
+resource mlProject 'Microsoft.MachineLearningServices/workspaces@2023-10-01' = {
```
```suggestion
resource project 'Microsoft.MachineLearningServices/workspaces@2023-10-01' = {
```

jongio commented on infra/app/ml.bicep:
```diff
@@ -0,0 +1,119 @@
+    type: 'SystemAssigned'
+  }
+  properties: {
+    friendlyName: contosoChatSfAIName
```
```suggestion
    friendlyName: contosoChatSfAiName
```

jongio commented on infra/app/ml.bicep:
```diff
@@ -0,0 +1,119 @@
+}
+
+resource openai 'Microsoft.CognitiveServices/accounts@2023-05-01' existing = {
+  name: openAIName
```
Keep consistent Ai instead of AI, check all files please.

jongio commented on infra/core/database/cosmos/sql/cosmos-sql-db.bicep:
```diff
@@ -0,0 +1,74 @@
+metadata description = 'Creates an Azure Cosmos DB for NoSQL account with a database.'
```
did you have to make any changes from azd/infra/core for this project?

zedy-wj replied:
Yes, we mainly modified the `container-registry` in core and configured [the creation of scopeMap](https://github.com/jongio/contoso-chat/pull/1/files#diff-82700b4bc57cc6f982085408349ebe12b24bcc0cdd7cc01bd2ce3d3a8da601f8R102-R105) resources for it, because we think this modification can be reused later.

As for other minor changes:
1. Added some default parameters in `container-registry`, such as `quarantinePolicy`, `trustPolicy`, etc. that were not available before.
2. Since some other resource IDs are needed when creating `MachineLearningServices`, these IDs are not output, so I added some required output IDs, such as [storageAccountId](https://github.com/jongio/contoso-chat/pull/1/files#diff-6c71e92c01838a849b8728b0368f0ac670f08310ccdd9df31fcab656f3567e19R88), [appInsightId](https://github.com/jongio/contoso-chat/pull/1/files#diff-e97bc2f50258546774b25a530dad18f715cf2fa070a9e91c474335114534d94cR31).

jongio replied:
Can you please do a PR to azd/infra/core in azure-dev repo for anything that has changed?

jongio commented on infra/hooks/prepenv.sh:
```diff
@@ -0,0 +1,38 @@
+echo "Loading azd .env file from current environment"
```
Please diff with my postprovision.sh script

jongio commented on infra/main.bicep:
```diff
@@ -1,19 +1,38 @@
+// }
+
+param applicationInsightsName string = ''
+param azureOpenAIResourceName string = ''
```
```suggestion
param azureOpenAiResourceName string = ''
```

jongio commented on infra/main.bicep:
```diff
@@ -1,19 +1,38 @@
+param storageServiceName string = ''
+
+param accountsContosoChatSfAIServicesName string = 'contoso-chat-sf-ai-aiservices'
+param workspacesApwsContosoChatSfAi362802272292Name string = 'apws-contosochatsfai362802272292'
```
Let's update this name to be something shorter.  Not sure why it was that long in the first plac.e

zedy-wj commented:
@jongio - Updated according to your comments. Please re-review it, thanks!

Regarding hooks, we used your original file(postprovision.sh) and modified some of the parameter names. And we added the steps to [install requirements](https://github.com/jongio/contoso-chat/pull/1/files#diff-b4dae3e1241bb856cb0942d1ffd5038172f42871ed7c12cd3931726cf71abb1aR77-R78).

jongio commented on azure.yaml:
```diff
@@ -3,12 +3,12 @@
-    interactive: true
-    run: infra/hooks/postprovision.sh
\ No newline at end of file
+    interactive: false
```
Why false?

jongio commented on infra/app/ml.bicep:
```diff
@@ -0,0 +1,120 @@
+param appInsightId string
+param containerRegistryId string
+param contosoChatSfAiName string = 'contoso-chat-sf-ai'
+param contosoChatSfAiprojName string = 'contoso-chat-sf-aiproj'
```
```suggestion
param contosoChatSfAiProjectName string = 'contoso-chat-sf-aiproj'
```

jongio commented on infra/app/ml.bicep:
```diff
@@ -0,0 +1,120 @@
+param appInsightId string
```
```suggestion
param applicationInsightsId string
```

jongio commented on infra/app/ml.bicep:
```diff
@@ -0,0 +1,120 @@
+    hbiWorkspace: false
+    v1LegacyMode: false
+    publicNetworkAccess: 'Enabled'
+    discoveryUrl: 'https://swedencentral.api.azureml.ms/discovery'
```
Let's use the location param for this.

jongio commented on infra/app/ml.bicep:
```diff
@@ -0,0 +1,120 @@
+  name: searchName
+}
+
+output workspace_name string = workspace.name
+output project_name string = project.name
+output workspace_principal_id string = workspace.identity.principalId
```
Note to check output casing

jongio commented on infra/main.bicep:
```diff
@@ -1,19 +1,38 @@
-// Necessary for GPT-4
-param searchLocation string = 'eastus'
-param resourceGroupName string = ''
+var openaiSubdomain = '${accountsContosoChatSfAiServicesName}${resourceToken}'
```
```suggestion
var openAiSubdomain = '${accountsContosoChatSfAiServicesName}${resourceToken}'
```

jongio commented on infra/main.bicep:
```diff
@@ -1,19 +1,38 @@
-param searchLocation string = 'eastus'
-param resourceGroupName string = ''
+var openaiSubdomain = '${accountsContosoChatSfAiServicesName}${resourceToken}'
+var openaiEndpoint = 'https://${openaiSubdomain}.openai.azure.com/'
```
```suggestion
var openAiEndpoint = 'https://${openaiSubdomain}.openai.azure.com/'
```

jongio commented on infra/main.bicep:
```diff
@@ -23,27 +42,319 @@ resource rg 'Microsoft.Resources/resourceGroups@2021-04-01' = {
-output search_name string = resources.outputs.search_name
-output mlhub_name string = resources.outputs.mlhub_name
-output mlproject_name string = resources.outputs.mlproject_name
+output AZURE_OPENAI_NAME string = openai.outputs.name
+output AZURE_COSMOS_NAME string = cosmos.outputs.accountName
+output AZURE_SEARCH_NAME string = search.outputs.name
+output AZURE_WORKSPACE_NAME string = machinelearning.outputs.workspace_name
+output AZURE_PROJECT_NAME string = machinelearning.outputs.project_name
```
Let's get consistent output casing.  I see camel and snake

zedy-wj commented:
@jongio - Updated according to your comments. Please re-review it, thanks!

jongio commented on infra/app/ml.bicep:
```diff
@@ -0,0 +1,119 @@
+    }
+  }
+
+  resource openaiConnection 'connections' = {
```
```suggestion
  resource openAiConnection 'connections' = {
```

jongio commented on infra/app/ml.bicep:
```diff
@@ -0,0 +1,119 @@
+    discoveryUrl: 'https://${location}.api.azureml.ms/discovery'
+  }
+
+  resource openaiDefaultEndpoint 'endpoints' = {
```
```suggestion
  resource openAiDefaultEndpoint 'endpoints' = {
```

jongio commented on infra/hooks/postprovision.sh:
```diff
@@ -43,16 +43,16 @@ fi
+cosmosService=$AZURE_COSMOS_NAME
 subscriptionId=$AZURE_SUBSCRIPTION_ID
-mlProjectName=$mlproject_name
+mlProjectName=$AZURE_PROJECT_NAME
```
```suggestion
mlProjectName=$AZURE_MLPROJECT_NAME
```

jongio commented on infra/hooks/postprovision.sh:
```diff
@@ -43,16 +43,16 @@ fi
 if [ -z "$resourceGroupName" ] || [ -z "$searchService" ] || [ -z "$openAiService" ] || [ -z "$cosmosService" ] || [ -z "$subscriptionId" ] || [ -z "$mlProjectName" ]; then
     echo "One or more required environment variables are not set."
-    echo "Ensure that AZURE_RESOURCE_GROUP, search_name, openai_name, cosmos_name, AZURE_SUBSCRIPTION_ID, and mlproject_name are set."
+    echo "Ensure that AZURE_RESOURCE_GROUP, AZURE_SEARCH_NAME, AZURE_OPENAI_NAME, AZURE_COSMOS_NAME, AZURE_SUBSCRIPTION_ID, and AZURE_PROJECT_NAME are set."
```
```suggestion
    echo "Ensure that AZURE_RESOURCE_GROUP, AZURE_SEARCH_NAME, AZURE_OPENAI_NAME, AZURE_COSMOS_NAME, AZURE_SUBSCRIPTION_ID, and AZURE_MLPROJECT_NAME are set."
```

jongio commented on infra/main.bicep:
```diff
@@ -23,28 +42,319 @@ resource rg 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+output AZURE_COSMOS_NAME string = cosmos.outputs.accountName
+output AZURE_SEARCH_NAME string = search.outputs.name
+output AZURE_WORKSPACE_NAME string = machinelearning.outputs.workspaceName
+output AZURE_PROJECT_NAME string = machinelearning.outputs.projectName
```
```suggestion
output AZURE_MLPROJECT_NAME string = machineLearning.outputs.projectName
```

jongio commented on infra/main.bicep:
```diff
@@ -23,28 +42,319 @@ resource rg 'Microsoft.Resources/resourceGroups@2021-04-01' = {
   }
 }
 
+module machinelearning 'app/ml.bicep' = {
```
```suggestion
module machineLearning 'app/ml.bicep' = {
```

jongio commented on infra/main.bicep:
```diff
@@ -23,28 +42,319 @@ resource rg 'Microsoft.Resources/resourceGroups@2021-04-01' = {
+output AZURE_OPENAI_NAME string = openai.outputs.name
+output AZURE_COSMOS_NAME string = cosmos.outputs.accountName
+output AZURE_SEARCH_NAME string = search.outputs.name
+output AZURE_WORKSPACE_NAME string = machinelearning.outputs.workspaceName
```
```suggestion
output AZURE_WORKSPACE_NAME string = machineLearning.outputs.workspaceName
```

zedy-wj commented:
> Can you please do a PR to azd/infra/core in azure-dev repo for anything that has changed?

@jongio - We have done a PR for anything changed in `infra/core`, please review it: https://github.com/Azure/azure-dev/pull/3495. In addition, your other comments have been processed, please re-review it, thanks!
